pypdf/filters.py

```diff
@@ -416,7 +413,7 @@ def next_code(self) -> int:
                     self.bytepos = self.bytepos + 1
             return value
 
-        def decode(self) -> str:
+        def decode(self) -> bytes:
```

Comment:
I disagree with this. `pypdf.filters` is a public class and thus possibly a breaking change, although there might not be many users of it.

Reply:
If people uses a filter function they most probably use `decode_stream_data()` as it does the job to select the good dissector. this function was returning `Union[str,bytes]` ; if you move it to `bytes` only the interchangeability is kept. Moreover it will simply their code as they won't have anymore to test between both types.

Discarding support to Python 3.7 is on discussion since a few months (with the advantage to allow walrus) .
We could put both for next release and move to 5.0.0

Reply:
This would not be compliant with our deprecation/change policy of public interfaces - they have to be announced in the previous major version.

Reply:
I repeat this is not a breaking changing considering the output of  `decode_stream_data()`. Just a caution should be written in the change log.
I do not even see any information to be added in the documentation as no pages deals with filters nor `decode_stream_data()`

Reply:
This probably could be a more general discussion, but the `pypdf.filters` module is public, although not part of the API docs. I would still consider this public API. Anyone who has used `LZWDecode().decode()` in a standalone fashion will see this as a breaking change as we are returning bytes instead of a string now. Strictly speaking, we would have to do our usual deprecation process by ensuring backwards compatibility for at least one further major version.